command/agent/job_endpoint.go

```diff
@@ -153,7 +153,7 @@ func (s *HTTPServer) jobPlan(resp http.ResponseWriter, req *http.Request,
 	}
 
 	requestRegion, jobRegion := regionForJob(
-		args.Job, args.WriteRequest.Region, s.agent.config.Region)
+		args.Job, args.Region, args.WriteRequest.Region, s.agent.config.Region)
```

Comment:
`args.Region` and `args.WriteRequest.Region` are the same variable; the latter is just explicitly accessed through the embedded struct `WriteRegion`

Reply:
the thing that you're looking for is in the `req` object:
https://github.com/hashicorp/nomad/blob/99529f233f827041d27b96a082b1a2b439c1c997/command/agent/http.go#L640

Reply:
Ok, that's a little weird the way that gets set... we're getting the query param for `args.Region` in [agent/http.go#L640](https://github.com/hashicorp/nomad/blob/master/command/agent/http.go#L640) and setting it as the [`args.Region`](https://github.com/hashicorp/nomad/blob/master/command/agent/job_endpoint.go#L30), whereas we set `args.WriteRequest.Region` in [agent/http.go#L680](https://github.com/hashicorp/nomad/blob/master/command/agent/http.go#L680). In which case it's not super clear to me when the request body would ever override the logic in [`parse_region`](https://github.com/hashicorp/nomad/blob/master/command/agent/http.go#L639)

But easy enough to drop the extra param.